### DISCUSSION TITLE
How to replace camera

### DISCUSSION THREAD

**Discussion by timoxley** [Sixze/ALS-Refactored#113] [Q&A]
I want to do two things:
1. Add children to the camera
2. Change the camera into a cine camera.

How can one get a hold of the camera component instance or edit the default settings?
It seems to be created by some magic I do not yet understand.
I've tried adding children to the Player's CameraManager, in BeginPlay but the children only appear when I enable the debug `;` camera. Seems the only camera settings that are exposed for customisation is FOV. edit: Ahh I've found the rest of the camera settings, but still don't see how to attach children.
Any pointers? I've searched blueprints and C++ for Camera, Camera Manager, Camera Component, etc, but didn't find anything that looked like it actually instantiated a camera.

**Comment by timoxley** [comment]
Ahh, ok the problem was that the camera manager is hidden in game, solution appears to be to: 

1. Add the children to the `Player Camera Manager` 
2. Set the `Player Camera Manager`'s `Actor Hidden in Game` to `false` e.g. in the `Player Controller`'s `BeginPlay`

![image](https://user-images.githubusercontent.com/43438/178575644-1285cd7f-6f0b-4bb4-96d7-dfdad04eae12.png)

I'm not sure if adding the child to the Camera Manager is "correct" or somehow different to adding it to a Camera Component, but seems to work, at least on the surface.

**Comment by timoxley** [comment]
I'm still curious whether there's a more direct, more "Normal" way of accessing and setting camera settings.

**Comment by Sixze** [comment]
ALS camera is just a separate self contained component like the standard camera components from the engine, it doesn't need `APlayerCameraManager`, `USpringArmComponent` or anything else. You can see an example of its usage inside `AAlsCharacterExample`.

**Comment by timoxley** [comment]
@Sixze sure, I looked at that but it inherits from skeletal mesh not camera?

**Comment by Alex-G** [comment]
> @Sixze sure, I looked at that but it inherits from skeletal mesh not camera?

Because it uses an animation blueprint to control the camera.

**Comment by timoxley** [comment]
How/where does it make a camera?

**Reply by Alex-G** [reply]
https://youtu.be/OEULrbcIQs4

**Reply by timoxley** [reply]
I've watched this and sure, I can see the camera in the ALS Camera Manager, which extends from Player Camera Manager, but no such thing exists in ALS Refactored. 
How and where does ALS Refactored interact with or instantiate an *actual* unreal camera?

**Reply by Sixze** [reply]
Take a look at the `AlsCharacterExample::CalcCamera()` function.
